### ISSUE TITLE
Some 0.8.x Projects are not Properly Converted into 1.x

### ISSUE THREAD

**Issue by noplay** [GNS3/gns3-gui#801]
Report by @jean-christophe-manciot

One example is the following source 0.8.3 project:
![image](https://cloud.githubusercontent.com/assets/345437/11058707/fb2a55c2-8794-11e5-8ff8-d6adaaad7d33.png)

... converted into the following destination 1.3.1 Linux project:
![image](https://cloud.githubusercontent.com/assets/345437/11058719/0ed49560-8795-11e5-9887-67b640d1eef8.png)

This lab is attached.
All the labs collection is available here (https://community.gns3.com/external-link.jspa?url=https%3A%2F%2Flearningnetwork.cisco.com%2Fthread%2F43019) to perform tests on a wide variety of labs.

Another example with this RIPng project converted from 0.8.5:
![image](https://cloud.githubusercontent.com/assets/345437/11058744/3d7a9ff4-8795-11e5-9bd4-f23961bc0e26.png)

... to 1.3.1/Linux:
![image](https://cloud.githubusercontent.com/assets/345437/11058749/45b89b58-8795-11e5-8248-573251d05130.png)

https://community.gns3.com/ideas/1757

**Comment by noplay** [comment]
Report by @jean-christophe-manciot:
I've just discovered other areas where conversion from 0.8.x projects falls short. The following is a comparison of a portion of .gns3 topology files after conversion with another native 1.3.7 (1.3.9 should lead to the same results) one:
After conversion from 0.8.x into 1.3.7:
{
                "description": "Router c3725",
                "dynamips_id": 3,
                "id": 3,
                "label": {
                    "color": "#000000",
                    "font": "TypeWriter,10,-1,5,75,0,0,0,0,0",
                    "text": "R1",
                    "x": 21.5,
                    "y": 20.0
                },
                "ports": [
                    {
                        "adapter_number": 0,
                        "description": "connected to R8 on port Serial0/0",
                        "id": 6,
                        "link_id": 8,
                        "name": "Serial0/0",
                        "nio": "NIO_UDP",
                        "port_number": 16
                    },
                    ...
                ],
                "properties": {
                    "aux": 2501,
                    "clock_divisor": 8,
                    "console": 2101,
                    "disk0": 16,
                    "disk1": 0,
                    "exec_area": 64,
                    "idlemax": 500,
                    "idlepc": "0x6062f0e0",
                    "idlesleep": 30,
                    "image": "c3725-adventerprisek9-mz.124-15.T10.image",
                    "iomem": 5,
                    "mac_addr": "c203.17db.0000",
                    "mmap": true,
                    "name": "R1",
                    "nvram": 112,
                    "platform": "c3725",
                    "ram": 512,
                    "slot0": "GT96100-FE",
                    "sparsemem": true,
                    "startup_config": "configs/i3_startup-config.cfg",
                    "system_id": "FTX0945W0MY",
                    "wic0": "WIC-2T"
                },
                "server_id": 1,
                "type": "C3725",
                "vm_id": "5de2bd25-c5da-4bd6-a227-ea4f431d2169",
                "x": -178.0,
                "y": -52.0
            },
 
Native 1.3.7:
{
                "default_symbol": ":/symbols/multilayer_switch.normal.svg",
                "description": "Router c3725",
                "dynamips_id": 1,
                "hover_symbol": ":/symbols/multilayer_switch.selected.svg",
                "id": 4,
                "label": {
                    "color": "#000000",
                    "font": "TypeWriter,10,-1,5,75,0,0,0,0,0",
                    "text": "ESW1",
                    "x": -3.828125,
                    "y": 50.0
                },
                "ports": [
                    {
                        "adapter_number": 1,
                        "description": "connected to vIOS-1 on port Ethernet1",
                        "id": 36,
                        "link_id": 4,
                        "name": "FastEthernet1/0",
                        "nio": "NIO_UDP",
                        "port_number": 0
                    },
                  ...
                ],
                "properties": {
                    "aux": 2501,
                    "clock_divisor": 8,
                    "console": 2004,
                    "disk0": 1,
                    "disk1": 0,
                    "exec_area": 64,
                    "idlemax": 500,
                    "idlepc": "0x6026b804",
                    "idlesleep": 30,
                    "image": "c3725-adventerprisek9-mz.124-15.T10.image",
                    "iomem": 5,
                    "mac_addr": "c201.1b91.0000",
                    "mmap": true,
                    "name": "ESW1",
                    "nvram": 256,
                    "platform": "c3725",
                    "private_config": "configs/i1_private-config.cfg",
                    "ram": 128,
                    "slot0": "GT96100-FE",
                    "slot1": "NM-16ESW",
                    "sparsemem": true,
                    "startup_config": "configs/i1_startup-config.cfg",
                    "system_id": "FTX0945W0MY",
                    "wic0": "WIC-2T"
                },
                "server_id": 1,
                "type": "C3725",
                "vm_id": "422d8521-dc33-44ed-852c-43dba27c88a1",
                "x": -548.0,
                "y": -92.0
            },
 
I believe there have been some changes in the format of the .gns3 file that have not been echoed in gns3-converter code...
 
When the 0.8.x project is converted, no 3725 node can start.
If you need this particular project to conduct your own tests, I can send it to you on demand.

**Comment by noplay** [comment]
Do you have a project which failed everytime? I don't have trouble with RIPNG and Transition through static tunnel :(

**Comment by jean-christophe-manciot** [comment]
What gns3/gns3-converter versions are you using?

**Comment by noplay** [comment]
Last from git

**Comment by jean-christophe-manciot** [comment]
I guess that your gns3-converter is more advanced than 1.2.3 then.
I have not mentioned that my tests have been conducted with the Linux version.
Could you continue your tests with the Linux labs (*-linux.net) behind the "Jean-Christophe" folders in CCNP/ROUTE & CCNP/SWITCH, one per major L2/L3 feature, for example one for BGP, one for EIGRP and so on?

**Comment by noplay** [comment]
Perhaps this commit has solve the issue:
https://github.com/GNS3/gns3-converter/commit/6fe9eef1c21d5611069fa5485826b73e36714234

**Comment by noplay** [comment]
I will test the other topologies also today or tomorrow.

**Comment by noplay** [comment]
Tested it seem to work.

**Comment by noplay** [comment]
I will see if daniel can release a new version

**Comment by jean-christophe-manciot** [comment]
I've just built the last gns3-converter 1.2.4.dev1 from sources.
None of my SWITCH labs pass the conversion, except the first one.

**Comment by noplay** [comment]
Can you send a sample lab by mail to julien@gns3.net ? Perhaps we are not
talking about the same labs.

**Comment by jean-christophe-manciot** [comment]
They all come from the link to the labs collection in post #1.

**Comment by jean-christophe-manciot** [comment]
But the patch you were talking about did improve the situation over some (ROUTE) labs, like the ones depicted in post #1.

**Comment by noplay** [comment]
I try with PVST+\ QOS/topology.net
![capture d ecran 2015-12-09 a 10 07 04](https://cloud.githubusercontent.com/assets/345437/11681126/2e008a5a-9e5d-11e5-8d06-3448891ba95a.png)


And PVST+\ EIGRP\ DHCP/topology-linux.net
![capture d ecran 2015-12-09 a 10 05 39](https://cloud.githubusercontent.com/assets/345437/11681129/3361d152-9e5d-11e5-82be-602cdeac5839.png)

And the result seem to match the screenshot. The error is later?

**Comment by jean-christophe-manciot** [comment]
Try labs n°2, 3, 4, 5 & 6. The conversion leads to a blank screen.

**Comment by noplay** [comment]
Ok the C7200-IO-FE was missing.

If you clone git@github.com:GNS3/gns3-converter.git and use the branch compatibility_fixes everything should work.

**Comment by jean-christophe-manciot** [comment]
Project n°4 (PVST+ EtherChannel OSPF) works now, but the other ones - 2, 3 , 5 & 6 - are still blank after conversion.

**Comment by noplay** [comment]
Which project are you calling n°2 ?

**Comment by jean-christophe-manciot** [comment]
2 - PVST+ EtherChannel HSRP
3 - PVST+ EtherChannel HSRP with Dual L2-L3 Links
5 - PVST+ EtherChannel VRRP
6 - PVST+ GLBP

**Comment by grossmj** [comment]
@noplay the main problem with these labs is to move and rename the nvram/disks where the VLAN information is stored, they have a different naming convention in 0.8.x (for instance c3725_R1_rom with R1 being the hostname) vs 1.x (for instance c3725_i1_rom, with i1 being the instance ID).

**Comment by jean-christophe-manciot** [comment]
But it's already done correctly with labs n°1, 4 & 7. They are the same as 2,3,5 & 6 in that regard.
What is different is the GNS3 version that was used to make these labs:
All the labs that pass the conversion:
1 - 0.8.3
4 - 0.8.4 RC2
7 - 0.8.3.1
The labs which cannot be converted:
2 - 0.8.6
3 - 0.8.6
5 - 0.8.6
6 - 0.8.6
What is special about 0.8.6 *.net file formats that is not understood by gns3-converter?

**Comment by noplay** [comment]
With lab 2:
![capture d ecran 2015-12-10 a 14 14 16](https://cloud.githubusercontent.com/assets/345437/11716203/6ebd9718-9f48-11e5-91fa-3414f6671ef3.png)

And I can start it :(

I will test on a different computer

**Comment by noplay** [comment]
Tested on a different computer and it's also work. If you stop gns3 between each .net import does it work? The problem seem deeper. Perhaps a race condition or something like that.

**Comment by jean-christophe-manciot** [comment]
Congrats! Now, with commit "Add missing C7200-IO-FE", all old SWITCH labs topologies are correctly converted. :+1: 
However, although all the nvram files seem there, no VLAN information seems to be loaded into the switches; could it be an id mismatch?

**Comment by noplay** [comment]
About the VLAN if have no idea. Do you have the correct name inside the routers?

**Comment by jean-christophe-manciot** [comment]
How about asking Jeremy. We discussed and solved a somewhat similar issue a few months ago where all the nvram files were deleted as soon as we exited GNS3.
If we could get access to some jungle discussions, I could show you the thread involved, but it seems they have vanished into thin air for now.
Regarding the name inside the 3725, there is no such thing in my labs, only numbers.
For example, in SWITCH/PVST+ EtherChannel HSRP, SW1 & SW2 have VLANs 1 <-> 5.

**Comment by noplay** [comment]
We have archive of community and can dig into it. It was in private message with @grossmj ?

**Comment by jean-christophe-manciot** [comment]
If I recall correctly, it was in 2 threads: 
- the first one on one of Jeremy's post about new GNS3 release, somewhere around 1.3.11/12, maybe a little before
- the second one on one of my threads, but I can't recall which one; it contains the keyword 3725

**Comment by grossmj** [comment]
> We discussed and solved a somewhat similar issue a few months ago where all the nvram files were deleted as soon as we exited GNS3.

This was fixed in 1.3.13 if I recall correctly.

Main issue is this https://github.com/GNS3/gns3-gui/issues/801#issuecomment-163360664 the disks should be moved and renamed by the converter.

**Comment by noplay** [comment]
Ok let's close this an reopen issue if we found something new or if someone found the message.

I opened a new issue for the nvram #958

And send a call to community to get feedback on what is not working:
https://gns3.com/discussions/-call-to-community-do-you-have-0

**Comment by jean-christophe-manciot** [comment]
@noplay I'm not sure why this thread has been closed since it is not solved completely.
@grossmj "Main issue is this #801 (comment) the disks should be moved and renamed by the converter." This has been fixed indeed and that's not what I am talking about.
Yet again, if I load one of my SWITCH labs (https://learningnetwork.cisco.com/thread/43019), for example PVST+ EtherChannel HSRP, all the files c3725_SWn_rom, c3725_SWn_slot0 & c3725_SWn_slot1 are converted into c3725_in_rom, c3725_in_slot0 & c3725_in_slot1 respectively, but on SW1 & SW2 for example, where VLANs 1 -> 5 were created on 0.8.7 lab, now they are gone:

SW1#sh vlan-sw

VLAN Name                             Status    Ports
---- -------------------------------- --------- -------------------------------
1    default                          active    Fa1/12, Fa1/13, Fa1/14, Fa1/15
1002 fddi-default                     act/unsup 
1003 token-ring-default               act/unsup 
1004 fddinet-default                  act/unsup 
1005 trnet-default                    act/unsup 

SW2#sh vlan-sw

VLAN Name                             Status    Ports
---- -------------------------------- --------- -------------------------------
1    default                          active    Fa1/12, Fa1/13, Fa1/14, Fa1/15
1002 fddi-default                     act/unsup 
1003 token-ring-default               act/unsup 
1004 fddinet-default                  act/unsup 
1005 trnet-default                    act/unsup 

So that means that **the files are not read correctly or some data are lost during the conversion**.

**Comment by jean-christophe-manciot** [comment]
I tried to close GNS3 and reload everything again, same result.

**Comment by noplay** [comment]
Hi,

Sorry for the misunderstanding. But I moved the NVRAM to a dedicated issue in order to clarify the situation and will open new issue if we identify other bug.

We think the problem of the VLAN is the NVRAM lost during conversion. The config are correctly copy but not the nvram. At this time the converter copy only the configs.

**Comment by jean-christophe-manciot** [comment]
Have you read https://github.com/GNS3/gns3-gui/issues/801#issuecomment-175083727?

**Comment by noplay** [comment]
Yes. The issue is we copy only the config and not the disk that why when you restart the topology the NVRAM is recreate via the startup config. But the vlan are not in startup config but in an other place of the NVRAM: vlan.dat.

After conversion of you see that the NVRAM file is missing, so the vlan information is lost:
```
ls -R
PVST+ EtherChannel HSRP-files PVST+ EtherChannel HSRP.gns3  topology-linux.png            topology.png

./PVST+ EtherChannel HSRP-files:
dynamips

./PVST+ EtherChannel HSRP-files/dynamips:
configs

./PVST+ EtherChannel HSRP-files/dynamips/configs:
i10_startup-config.cfg i12_startup-config.cfg i2_startup-config.cfg  i4_startup-config.cfg  i6_startup-config.cfg  i8_startup-config.cfg
i11_startup-config.cfg i1_startup-config.cfg  i3_startup-config.cfg  i5_startup-config.cfg  i7_startup-config.cfg  i9_startup-config.cfg

**Comment by jean-christophe-manciot** [comment]
OK, the files c3725_in_rom, c3725_in_slot0 & c3725_in_slot1 **appear in dynamips folder only after we start the corresponding nodes, not right after the conversion**. My bad. Hence the missing VLANs...

**Comment by noplay** [comment]
Could you check if everything is OK with this conversion:
[topology.zip](https://github.com/GNS3/gns3-gui/files/106766/topology.zip)

I will make a PR to GNS3 converter

**Comment by jean-christophe-manciot** [comment]
Nothing works as expected; on SW1, all the configuration is gone now:

         --- System Configuration Dialog ---

Would you like to enter the initial configuration dialog? [yes/no]: 
% Please answer 'yes' or 'no'.

**Comment by noplay** [comment]
Hmm something seem to have been corrupted when zipped.

I rexport it:
https://framadrop.org/r/Hd4QLMvz3j#Z1XwPFEMLj8JREPnVBr6ztcwBN+ttghedmBl+2qyyyc=

The result for me:
```
SW1>sh vlan-sw

VLAN Name                             Status    Ports
---- -------------------------------- --------- -------------------------------
1    default                          active    Fa1/0, Fa1/1, Fa1/2, Fa1/3
                                                Fa1/4, Fa1/5, Fa1/6, Fa1/7
                                                Fa1/8, Fa1/9, Fa1/10, Fa1/11
                                                Fa1/12, Fa1/13, Fa1/14, Fa1/15
1002 fddi-default                     active    
1003 token-ring-default               active    
1004 fddinet-default                  active    
1005 trnet-default                    active    

VLAN Type  SAID       MTU   Parent RingNo BridgeNo Stp  BrdgMode Trans1 Trans2
---- ----- ---------- ----- ------ ------ -------- ---- -------- ------ ------
1    enet  100001     1500  -      -      -        -    -        1002   1003
1002 fddi  101002     1500  -      -      -        -    -        1      1003
1003 tr    101003     1500  1005   0      -        -    srb      1      1002
1004 fdnet 101004     1500  -      -      1        ibm  -        0      0   
1005 trnet 101005     1500  -      -      1        ibm  -        0      0

**Comment by jean-christophe-manciot** [comment]
The probable issue is that you've loaded the lab with a different image 3640 vs 3725.
Since GNS3 produces deltas from the base image, the result of loading the lab with a different base image is unpredictable. So you really need the same 3725 image.
Anyway, the VLANs 2->5 are still not available. What you see 1002 -> 1005 are default & different VLANs.

**Comment by noplay** [comment]
Is it better?
https://framadrop.org/r/AA-xig8gnI#Otny3GLNrdb0aq3u4YbJCFXpCIn1QA1ImEWeQQCiU94=
```
SW1#vl

VLAN Name                             Status    Ports
---- -------------------------------- --------- -------------------------------
1    default                          active    Fa1/12, Fa1/13, Fa1/14, Fa1/15
2    VLAN0002                         active    
3    VLAN0003                         active    
4    VLAN0004                         active    
5    VLAN0005                         active    
99   VLAN0099                         active    
1002 fddi-default                     act/unsup 
1003 trcrf-default                    act/unsup 
1004 fddinet-default                  act/unsup 
1005 trbrf-default                    act/unsup

**Comment by jean-christophe-manciot** [comment]
Yes, **much better** :-) I cross-checked with the original 0.8.7 version and the information contained within vlan.dat (which stores VLANs & VTP data) seems exported correctly between both versions.

However, there are a few strange things: 
- why is the folder tree style from 1.0 version instead of 1.4?
- when reading the topology file with GNS3 1.4, some dialog windows pop up to offer different images because their paths have changed since then; that's perfect, but after applying the changes, the results are random: sometimes, all the nodes are configured with c7200 image, sometimes with c3640 instead of c7200 & c3725... It may have something to do with the fact that the .gns3 file was designed for GNS3/Windows, but I'm reading it on GNS3/Linux.

**Comment by noplay** [comment]
Converter as not been updated for direct conversion to 1.4, but GNS3 already manage it when you load the project.

**Comment by imyelmo** [comment]
I'm having problems with certain topology during conversion. Is there any way to figure out what's going on?

**Comment by jean-christophe-manciot** [comment]
How about describing these issues so that @noplay or @grossmj can help you out and solve them?

**Comment by noplay** [comment]
@jean-christophe-manciot is right in order to help you I suggest you open a new issue with details:
* the topology
* screenshot before conversion
* screenshot after conversion

**Comment by noplay** [comment]
Sorry I just see and reply:
https://github.com/GNS3/gns3-converter/issues/6
